From e69899b984b70b90f22ef28138277e66a9ad2884 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Wed, 26 Aug 2020 00:02:19 +0300
Subject: [PATCH 14/14] rsi91x: Check wowlan pointer in rsi_shutdown

Check wowlan pointer before calling rsi_config_wowlan to
prevent erroneous configuration attempts

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 drivers/net/wireless/rsi/rsi_91x_sdio.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rsi/rsi_91x_sdio.c b/drivers/net/wireless/rsi/rsi_91x_sdio.c
index 1bebba4e8527..1fd208bea839 100644
--- a/drivers/net/wireless/rsi/rsi_91x_sdio.c
+++ b/drivers/net/wireless/rsi/rsi_91x_sdio.c
@@ -1473,7 +1473,7 @@
 	if (hw) {
 		struct cfg80211_wowlan *wowlan = hw->wiphy->wowlan_config;

-		if (rsi_config_wowlan(adapter, wowlan))
+		if (wowlan && rsi_config_wowlan(adapter, wowlan))
 			rsi_dbg(ERR_ZONE, "Failed to configure WoWLAN\n");
 	}

-- 
2.26.2

